### PR TITLE
fix(thorvg): fix clipped clippers

### PR DESCRIPTION
**Pull request by FASTSHIFT**
Mirror fixed: https://github.com/thorvg/thorvg/pull/2778

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/libs/thorvg/tvgPaint.cpp [hunk 1]
```diff
@@ -268,7 +268,7 @@ RenderData Paint::Impl::update(RenderMethod* renderer, const RenderTransform* pT
             }
             if (tryFastTrack) {
                 RenderRegion viewport2;
-                if ((compFastTrack = _compFastTrack(target, pTransform, target->pImpl->rTransform, viewport2)) == Result::Success) {
+                if (compData->method != CompositeMethod::ClipPath && (compFastTrack = _compFastTrack(target, pTransform, target->pImpl->rTransform, viewport2)) == Result::Success) {
```
Comment (hermet) [inline src/libs/thorvg/tvgPaint.cpp]: @FASTSHIFT Please try with this:

```
if (target->composite(nullptr) != CompositeMethod::ClipPath)  && (compFastTrack = _compFastTrack(target, pTransform, target->pImpl->rTransform, viewport2)) == Result::Success) {
```

Comment (FASTSHIFT) [reply src/libs/thorvg/tvgPaint.cpp]: Thank you. It's proven to work well.

**Review by kisvegabor — APPROVED** [review]

**Comment by liamHowatt** [comment]
Normally when we update ThorVG in LVGL, we overwrite the full contents of `src/libs/thorvg` with the full source tree of the new thorvg. The suggested implementation of the fix by @hermet will get lost the next time we update thorvg. What should we do about that?

**Comment by liamHowatt** [comment]
Thanks @hermet, my concern is that the implementation of the fix in this PR is different than the way the fix was implemented upstream in https://github.com/thorvg/thorvg/pull/2778. You suggested that @FASTSHIFT implement it [this way](https://github.com/lvgl/lvgl/pull/6956#discussion_r1779461626) in this PR, and they did. However, it will get overwritten by the upstream implementation eventually. Is the upstream implementation equally effective at fixing the clipped corners and fine to overwrite yours sometime later? Or is your implementation better somehow? Should another PR be opened upstream?

**Comment by hermet** [comment]
They are equally effective. Just internal interfaces diffs due to the different versions.

**Review by liamHowatt — APPROVED** [review]
Okay, thanks!

**Pull request merged**